native/src/main/java/io/ballerina/stdlib/java.jms/listener/ListenerImpl.java

```diff
@@ -71,17 +70,19 @@ public void onMessage(Message message) {
                     module.getOrg(), module.getName(), module.getVersion(), SERVICE_RESOURCE_ON_MESSAGE);
             ObjectType serviceType = (ObjectType) TypeUtils.getReferredType(TypeUtils.getType(consumerService));
             Object[] params = methodParameters(serviceType, message);
+            Object result;
             if (serviceType.isIsolated() && serviceType.isIsolated(SERVICE_RESOURCE_ON_MESSAGE)) {
-                ballerinaRuntime.invokeMethodAsyncConcurrently(
-                        consumerService, SERVICE_RESOURCE_ON_MESSAGE, null, metadata, callback,
-                        null, PredefinedTypes.TYPE_NULL, params);
+                result = ballerinaRuntime.startIsolatedWorker(
```

Comment:
Should we use new Virtual thread since this will block the current thread?

Reply:
Hope this works 3b97839f45f16295e5c3d962f2cce4153cf92174